test/integration/header_integration_test.cc

```diff
@@ -34,7 +34,7 @@ std::string ipSuppressEnvoyHeadersTestParamsToString(
 void disableHeaderValueOptionAppend(
```

Comment:
I would add an integration test for the new behavior.

Reply:
Added, Thanks!